### PR TITLE
[pick-7.5]region_request: ignore resource group errors that not relative storage layer (#1354)

### PR DESCRIPTION
**Pull request by nolouch**
ref https://github.com/tikv/client-go/issues/1322

### CODE REVIEW

File: internal/locate/region_request_test.go [hunk 1]
```diff
@@ -827,3 +864,20 @@ func (s *testRegionRequestToSingleStoreSuite) TestRegionRequestStats() {
 	}
 	s.Contains(expecteds, access.String())
 }
+
+type noCauseError struct {
```
Comment (cfzjywxk) [inline internal/locate/region_request_test.go:868]: @nolouch 
Unrelated changes are included.

Comment (nolouch) [reply internal/locate/region_request_test.go:868]: cc @cfzjywxk 
should pick https://github.com/tikv/client-go/pull/1344, otherwise, will panic.

**Review by cfzjywxk — APPROVED** [review]

**Review by ekexium — APPROVED** [review]
